libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -112,17 +119,17 @@ bool IRGeneratorForStatements::visit(Assignment const& _assignment)
 	solUnimplementedAssert(_assignment.assignmentOperator() == Token::Assign, "");
 
 	_assignment.rightHandSide().accept(*this);
+	Type const* intermediateType = _assignment.rightHandSide().annotation().type->closestTemporaryType(
+		_assignment.leftHandSide().annotation().type
+	);
+	string intermediateValue = expressionAsType(_assignment.rightHandSide(), *intermediateType);
 
-	// TODO proper lvalue handling
-	auto const& lvalue = dynamic_cast<Identifier const&>(_assignment.leftHandSide());
-	string varName = m_context.variableName(dynamic_cast<VariableDeclaration const&>(*lvalue.annotation().referencedDeclaration));
+	_assignment.leftHandSide().accept(*this);
```

Comment:
Should this maybe be done before line 123?

Reply:
You mean before we compute the intermediate type? This is the code generator, so the types are already fixed - or why do you think it should be done earlier?

Reply:
I know that in the `FunctionCall` visitor we ran `accept()` for every argument before talking a closer look at it, that's where my thought came from

Reply:
Actually I'm wondering whether it might make sense to never call `accept` explicitly in the yul code generation, i.e. only use the `endVisit` functions. For sol -> evm the order was kind of important because it determined the order on the stack, but now this is irrelevant.

Reply:
I'm thinking that too. I think using only `endVisit` would look nicer.